design/batch_retrieval_design.md

```diff
@@ -2,19 +2,140 @@
+The following table outlines possible solutions to overcome this:
 
-|      | Solution                                                     | Pros                                                         | Cons                                                         | Effort (T-shirt size) |
-| ---- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ | --------------------- |
-| 1    | *Client side:* <br />For each K8s Secret, perform a new Batch request | - No server side changes / no breaking changes               | Load on server with extra calls (*)                          | S, 5 days             |
-| 2    | *Server side:*<br />Update Batch retrieval to return list of variables **and** their response (success/failure) (**) | - Will help us during rotation for Milestone 2<br />- Better / straightforward design for how batch endpoints | - Requires both client/server implementation<br />- Need to handle backwards compatibility | M, 10 days            |
-| 3    | Stay as is                                                   | No additional work needed                                    | Bad UX                                                       | -                     |
+|      | Solution                                                     | Pros                                                         | Cons                                               | Effort (T-shirt size) |
+| ---- | ------------------------------------------------------------ | ------------------------------------------------------------ | -------------------------------------------------- | --------------------- |
+| 1    | *Client side:* <br />For each K8s Secret, perform a new Batch request | - No server side changes / no breaking changes               | Load on server with extra calls (*)                | S, 5 days             |
+| 2    | *Server side:*<br />Update Batch retrieval to return list of variables **and** their response (success/failure) (**) | - Will help us during rotation for Milestone 2<br />- Best practice / straightforward design for batch endpoint | - Requires both client/server implementation<br /> | M, 10 days            |
+| 3    | Stay as is                                                   | No additional work needed                                    | Bad UX                                             | -                     |
 
 (*) Fallback solution: use this solution as a safety net, only if the original Batch Retrieval request fails
 
 (**) This solution can be broken up in two: 1. Create a new API endpoint or 2. use existing one (return 409 response)
 
+*Decision:* We decided to go with #2 and create our own API. The new batch retrieval API endpoint will not be forward facing and be for **internal use only**. 
```

Comment:
so the new endpoint will be fully developed but it won't be in the docs? other than not being in the docs, will it have any difference from the other endpoints?

Reply:
also - why isn't it customer facing?

Reply:
We originally decided not to document because we already have a batch retrieval endpoint and so introducing another one is confusing because on paper, not much is different expect partial success. We thought to add onto the existing batch endpoint but that would lead to changing behaviors for our needs which is not what we wanted to accomplish

@abrahamko mentioned that from what he read it is better to document, even if there is duplication involved.

After convos on slack, it seems that this will be revisited. I will add this to an open question in the doc

Reply:
There was no explicit request from any customer to have additional format of batch retrieval  -and it is basically wrong to provide 2 different ways of doing things

Reply:
@oburstein-hub are you saying we recommend against creating an additional api endpoint?

Reply:
I am saying that there should be only one endpoint exposed to customers